nixos/modules/config/qt5.nix

```diff
@@ -0,0 +1,102 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.qt5;
+
+  determinePackages = platform: style:
+    if (platform == "gnome" && style == "adwaita") then [ pkgs.qgnomeplatform pkgs.adwaita-qt ]
+    else if (platform == "gtk2" && style != "adwaita") then [ pkgs.qtstyleplugins ]
+    else throw "`qt5.platformTheme` ${platform} and `qt5.style` ${style} are not compatable.";
+
+  packages = determinePackages cfg.platformTheme cfg.style;
```

Comment:
And I personally would inline `determinePackages`, there's not much to be gained from making this a function when it's only called once

Reply:
I just wanted it to be easy to expand with more conditions, how about?
```diff
diff --git a/nixos/modules/config/qt5.nix b/nixos/modules/config/qt5.nix
index 91f1adca220..5f806b113e4 100644
--- a/nixos/modules/config/qt5.nix
+++ b/nixos/modules/config/qt5.nix
@@ -6,12 +6,12 @@ let
 
   cfg = config.qt5;
 
-  determinePackages = platform: style:
-    if (platform == "gnome" && style == "adwaita") then [ pkgs.qgnomeplatform pkgs.adwaita-qt ]
-    else if (platform == "gtk2" && style != "adwaita") then [ pkgs.qtstyleplugins ]
-    else throw "`qt5.platformTheme` ${platform} and `qt5.style` ${style} are not compatable.";
+  isQGnome = cfg.platformTheme == "gnome" && cfg.style == "adwaita";
+  isQtStyle = cfg.platformTheme == "gtk2" && cfg.style != "adwaita";
 
-  packages = determinePackages cfg.platformTheme cfg.style;
+  packages = if isQGnome then [ pkgs.qgnomeplatform pkgs.adwaita-qt ]
+    else if isQtStyle then [ pkgs.qtstyleplugins ]
+    else throw "`qt5.platformTheme` ${cfg.platformTheme} and `qt5.style` ${cfg.style} are not compatable.";
 
 in
 
```

Reply:
I don't entirely get why those conditions have to be like this as I don't have any experience with QT stuff, but I like the code